### ISSUE TITLE
Object deletion by scope

### ISSUE THREAD

**Issue by kanes115** [stavro/arc#294]
It would be nice to be able to delete files by scope (and not by a specific file name). There are use cases where you upload multiple files per scope and want to delete every file that is connected to it at some point. If that is somehow possible now, please refer me to a place in docs/code.

### Environment

* Elixir version (elixir -v): 1.9.1
* Arc version (mix deps):
  arc_gcs 0.2.1 (Hex package) (mix)
  locked at 0.2.1 (arc_gcs) 7b90e69f
* Operating system: MacOS

**Comment by achempion** [comment]
You mean, you want to have files deleted automatically as a callback of deleting base resource?

**Comment by kanes115** [comment]
@achempion yes, all files connected with a given `scope`. Not necessarily on a callback (could be explicit).
